lib/expand/join.go

```diff
@@ -150,18 +150,23 @@ func (p *Peer) Execute(req *installpb.ExecuteRequest, stream installpb.Agent_Exe
 		case event := <-p.dispatcher.Chan():
 			err := stream.Send(event)
 			if err != nil {
-				return trace.Wrap(err)
+				return err
 			}
 		case req := <-p.closeC:
 			err := stream.Send(req.resp)
 			close(req.doneC)
 			if err != nil {
-				return trace.Wrap(err)
+				return err
 			}
 		case err := <-p.execDoneC:
 			if err != nil {
-				// Phase finished with an error
-				return status.Error(codes.FailedPrecondition, err.Error())
+				// Phase finished with an error.
+				if installpb.IsRPCError(err) {
+					return trace.Unwrap(err)
```

Comment:
I need to relay RPC errors to client unchanged since client has specific logic around specific error types and this gets messed up if we wrap the errors at RPC API boundary with `trace.Wrap`.

Reply:
It is the "boundary" function.